### PR TITLE
refactor(lsp): move changetracking to separate file

### PR DESCRIPTION
**Pull request by lewis6991**
- Prefixed changetracking types with `vim.lsp.`

### CODE REVIEW

File: runtime/lua/vim/lsp.lua [hunk 1]
```diff
@@ -132,9 +132,10 @@ local format_line_ending = {
   ['mac'] = '\r',
 }
 
+---@private
 ---@param bufnr (number)
 ---@return string
-local function buf_get_line_ending(bufnr)
+function lsp._buf_get_line_ending(bufnr)
```
Comment (lewis6991) [inline runtime/lua/vim/lsp.lua:138]: This and `buf_get_full_text` probably makes more sense to have in `lsp.util`?

Comment (clason) [reply runtime/lua/vim/lsp.lua:138]: No, `lsp.util` *delendam est*. Can't we find a better, permanent, place for it?

Comment (lewis6991) [reply runtime/lua/vim/lsp.lua:138]: `lsp._util`?

**Review by mfussenegger — APPROVED** [review]
lgtm, I assume the motivation is to have a smaller lsp.lua?

**Comment by lewis6991** [comment]
Yes. Seems like an obvious thing to have in a separate file instead of being 300 line `do`-`end` block.

Ideally `lsp.lua` should just contain top level stuff similar to `treesitter.lua`.

**Pull request merged**